src/main/scala/net/kaliber/play/prerender/PrerenderActionBuilders.scala

```diff
@@ -40,13 +40,25 @@ case class PrerenderActionBuilders(config: PrerenderConfig)(implicit ec: Executi
   }
 
   private def shouldBePrerendered(request: RequestHeader) =
-    config.enabled && isGetRequest(request) && isSearchEngineRequest(request)
+    config.enabled && isGetRequest(request) && isSearchEngineRequest(request) && hasOnlyAllowedParams(request) && !isExcludedWithRegex(request)
 
   private def isGetRequest(request: RequestHeader) = request.method == "GET"
 
   private def isSearchEngineRequest(request: RequestHeader) =
     isEscapedFragmentUrl(request) || hasSearchEngineUserAgent(request)
 
+  private def hasOnlyAllowedParams(request: RequestHeader) = {
+    val requestParams = request.queryString.keys.toSet
+    requestParams subsetOf config.limitedParams.getOrElse(requestParams)
+  }
+
+  private def isExcludedWithRegex(request: RequestHeader) = {
+    config.excludeRegex.getOrElse("").r.findFirstMatchIn(request.uri) match {
+      case None => false
+      case Some(matchingSubstring) => !matchingSubstring.toString().isEmpty
+    }
```

Comment:
About your approach. There is two parts that raise a red flag in my programmers brain:

1. Removing the `Option` with `getOrElse("")` and then continue to make a regular expression from it and executing it.
2. Returning `false` (an explicit boolean) for `None`, `Option` is inherently a boolean like type.

In any case. Tomorrow I am at at my place of work where I have the required permissions to release a new version. I will change this part and release. I'll let you know as soon as I have released it. Thank you for your contribution.